pkg/tenant/topics.go

```diff
@@ -69,68 +70,101 @@ func (s *Server) ProjectTopicCreate(c *gin.Context) {
-	var orgID ulid.ULID
-	if orgID, err = ulid.Parse(claims.OrgID); err != nil {
-		log.Error().Err(err).Msg("could not parse org id")
-		c.JSON(http.StatusInternalServerError, api.ErrorResponse("could not parse org id"))
+	// Get user credentials to make request to Quarterdeck.
+	if ctx, err = middleware.ContextFromRequest(c); err != nil {
+		log.Error().Err(err).Msg("could not create user context from request")
+		c.JSON(http.StatusUnauthorized, api.ErrorResponse("could not fetch user credentials"))
 		return
 	}
 
-	// Get project ID from the URL and return a 400 response if it is missing.
-	var projectID ulid.ULID
-	if projectID, err = ulid.Parse(c.Param("projectID")); err != nil {
-		log.Error().Err(err).Msg("could not parse project ulid")
-		c.JSON(http.StatusBadRequest, api.ErrorResponse("could not parse project id"))
+	// Fetch user claims from the context.
+	if claims, err = middleware.GetClaims(c); err != nil {
+		log.Error().Err(err).Msg("could not fetch user claims")
+		c.JSON(http.StatusUnauthorized, api.ErrorResponse("could not fetch claims from context"))
 		return
 	}
 
-	// Bind the user request with JSON and return a 400 response
-	// if binding is not successful.
+	// Bind the user request with JSON and return a 400 response if binding is not successful.
 	if err = c.BindJSON(&topic); err != nil {
 		log.Warn().Err(err).Msg("could not bind project topic create request")
 		c.JSON(http.StatusBadRequest, api.ErrorResponse("could not bind request"))
 		return
 	}
 
-	// Verify that a topic ID does not exist and return a 400 response
-	// if the topic ID exists.
+	// Verify that a topic ID does not exist and return a 400 response if the topic ID exists.
 	if topic.ID != "" {
 		c.JSON(http.StatusBadRequest, api.ErrorResponse("topic id cannot be specified on create"))
 		return
 	}
 
-	// Verify that a topic name has been provided and return a 400 response
-	// if the topic name does not exist.
+	// Verify that a topic name has been provided and return a 400 response if the topic name does not exist.
 	if topic.Name == "" {
 		c.JSON(http.StatusBadRequest, api.ErrorResponse("topic name is required"))
 		return
 	}
 
-	t := &db.Topic{
-		OrgID:     orgID,
-		ProjectID: projectID,
-		Name:      topic.Name,
+	// Get project ID from the URL and return a 404 response if it is missing.
+	var projectID ulid.ULID
+	if projectID, err = ulid.Parse(c.Param("projectID")); err != nil {
+		log.Error().Err(err).Str("projectID", projectID.String()).Msg("could not parse project ulid")
+		c.JSON(http.StatusNotFound, api.ErrorResponse("project not found"))
+		return
 	}
 
-	// Add topic to the database and return a 500 response if not successful.
-	if err = db.CreateTopic(c.Request.Context(), t); err != nil {
-		log.Error().Err(err).Msg("could not create project topic in the database")
-		c.JSON(http.StatusInternalServerError, api.ErrorResponse("could not add project topic"))
```

Comment:
Is there a reason we deleted saving the metadata to the trtl database?

Reply:
I was thinking of returning the data directly from Ensign, but now I realize that this is a skipped step and it should be added back. Adding this back should then save the data in trtl, then what goes out will have the data from trtl. So, I think that answers your other question.

Reply:
Sorry, I think I jumped ahead - ping me when you're ready for me to review!